src/python/python/dict_to_xml.py

```diff
@@ -0,0 +1,615 @@
+                self.write_dict(params)
+                self.close_element()
+            else:
+                raise ValueError("Unsupported entry: (%s,%s)" % (key,value))
+
+        if len(self.file_stack[self.current_file]) == 1:
+            #close scene tag
+            self.close_element()
+
+    def configure(self, scene_dict):
+        '''
+        Special handling of configure API.
+        '''
+        self.preprocess_scene(scene_dict.copy()) # Re order elements
+        #TODO: open files only here
+        if self.split_files:
+            for file in [Files.MAIN, Files.CAMS, Files.MATS, Files.GEOM, Files.EMIT]:
+                self.set_output_file(file)
+                self.write_dict(self.scene_data[file])
+        else:
+            self.write_dict(self.scene_data[Files.MAIN])
+
+        # Close files
+        print('Wrote scene files.')
+        for f in self.files:
+            if f is not None:
+                f.close()
+                print(' %s' % f.name)
+
+    def cleanup(self):
+        self.exit()
+
+    def exit(self):
+        # If any files happen to be open, close them and start again
+        for f in self.files:
+            if f is not None:
+                f.close()
+
+    def transform_matrix(self, transform):
+        '''
+        Converts a mitsuba Transform4f into a dict entry
+        '''
+
+        value = " ".join(["%f" % x for x in transform.matrix.numpy().flatten()])
+
+        params = {
+            'matrix': {
+                'type': 'matrix',
+                'value': value,
+            }
+        }
+        return params
+
+    def decompose_transform(self, transform, export_scale = False):
+        '''
+        Export a transform as a combination of rotation, scale and translation.
+        This helps manually modifying the transform after export (for cameras for instance)
+        '''
+        transform = transform.matrix.numpy()
+
+        params = {}
+
+        tr = transform[:3,3]
+        scale = [np.linalg.norm(transform[:3,j]) for j in range(3)]
+        rot_mat = transform[:3, :3]
+        for j in range(3):
+            rot_mat[:3,j] *= 1.0 / scale[j]
+        rot = mat_to_euler(rot_mat)
+
+        params['rotate_x'] = {
+            'type': 'rotate',
+            'x': '1',
+            'angle': rot[0] * 180 / np.pi
+        }
+        params['rotate_y'] = {
+            'type': 'rotate',
+            'y': '1',
+            'angle': rot[1] * 180 / np.pi
+        }
+        params['rotate_z'] = {
+            'type': 'rotate',
+            'z': '1',
+            'angle': rot[2] * 180 / np.pi
+        }
+        params['translate'] = {
+            'type': 'translate',
+            'value': "%f %f %f" % tuple(tr)
+        }
+        if export_scale:
+            params['scale'] = { #TODO: remove this for cameras
+                'type': 'scale',
+                'value': "%f %f %f" % tuple(scale)
+            }
+
+        return params
+
+def mat_to_euler(mat):
```

Comment:
This function could probably be useful to other developers. Doesn't it already exist in enoki? Otherwise it should either be added to `enoki` or we could implement this in `C++` and provide bindings for it.